x-pack/test/functional/apps/ml/data_visualizer/index_data_visualizer_grid_in_dashboard.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { TestData, MetricFieldVisConfig } from './types';
+import { farequoteLuceneFiltersSearchTestData } from './index_test_data';
+
+const SHOW_FIELD_STATISTICS = 'discover:showFieldStatistics';
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const PageObjects = getPageObjects(['common', 'discover', 'timePicker', 'settings', 'dashboard']);
+  const ml = getService('ml');
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const toasts = getService('toasts');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+
+  const selectIndexPattern = async (indexPattern: string) => {
+    await retry.tryForTime(2 * 1000, async () => {
+      await PageObjects.discover.selectIndexPattern(indexPattern);
+      const indexPatternTitle = await testSubjects.getVisibleText('indexPattern-switch-link');
+      expect(indexPatternTitle).to.be(indexPattern);
+    });
+  };
+
+  const clearAdvancedSetting = async (propertyName: string) => {
```

Comment:
We have the same helper method in `index_data_visualizer_grid_in_discover.ts`. Would probably be good to move this to a service, e.g. to `test_resources.ts`. I also thought about maybe not using the UI to change the advanced settings, but the API, similar to how we already [set and reset the Kibana time zone](https://github.com/elastic/kibana/blob/main/x-pack/test/functional/services/ml/test_resources.ts#L33-L41).

Reply:
Updated here https://github.com/elastic/kibana/pull/116774/commits/528287c1182db01da221c740f3be3d738d046971